env/env.cc

```diff
@@ -693,6 +696,81 @@ Status Env::GetHostNameString(std::string* result) {
   return s;
 }
 
+namespace {
+
+// For GenerateRawUuid()
+struct RawUuidHelper {
+  // All the data we want to go into the initial random seed. For safety in
+  // case of API issues, we have two entropy tracks that should each be
+  // sufficient for generating unique ids.
+  struct Init {
+    // Track 1: host and thread identifiers and time
+    // TODO? process ID
+    uint64_t thread_id = 0;
+    std::array<char, 64> hostname_buf{};
+    int64_t time = 0;
+    uint64_t nano_time = 0;
+
+    // Track 2: random_device
+    using RandType = std::random_device::result_type;
+    static constexpr size_t kNumRandVals =
+        /* generous bits */ 192U / (8U * sizeof(RandType));
+    std::array<RandType, kNumRandVals> rand_vals{};
+  };
+
+  RawUuidHelper(Env* env) {
+    Init init{};
+
+    // Track 1: host and thread identifiers and time
+    init.thread_id = env->GetThreadID();
+    env->GetHostName(init.hostname_buf.data(), init.hostname_buf.size())
+        .PermitUncheckedError();
+    env->GetCurrentTime(&init.time).PermitUncheckedError();
+    init.nano_time = env->NowNanos();
+
+    // Track 2: random_device
+    std::random_device r;
+    for (auto& val : init.rand_vals) {
+      val = r();
+    }
+
+    // Mix it all together (nom nom)
+    state = XXH128(&init, sizeof(init), /*seed*/ 123U);
+  }
+
+  XXH128_hash_t state;
+};
+
+}  // namespace
+
+RawUuid Env::GenerateRawUuid() {
```

Comment:
I am a little confused as to if this code belongs here or is part of the Env Posix class (it looks like Windows may have a different implementation)?